programs/stake/src/stake_instruction.rs

```diff
@@ -552,6 +561,14 @@ mod tests {
                     config::create_account(0, &config::Config::default())
                 } else if sysvar::rent::check_id(&meta.pubkey) {
                     account::create_account(&Rent::default(), 1)
+                } else if meta.pubkey == bad_stake_pubkey() {
+                    let mut account = Account::default();
+                    account.owner = id();
+                    account
+                } else if meta.pubkey == bad_vote_pubkey() {
+                    let mut account = Account::default();
+                    account.owner = solana_vote_program::id();
+                    account
```

Comment:
These are for test cases where the owner should be correct but the account data can't be deserialized as StakeState/VoteState respectively, correct?
I found that a little confusing in the context of the rest of the PR, which deals with valid accounts with bad owners.
I think renaming the pubkey functions a little would help... I'll suggest above

Reply:
thanks for the suggestion!: done: https://github.com/solana-labs/solana/pull/13615/commits/933dcfd465775638f63cc323227d0b40cd6d55d6